bin/BuildPackages.sh

```diff
@@ -201,10 +201,16 @@ GAPInput
 )
       local CONFIG_ARGS_FLAG_NAME="PACKAGE_CONFIG_ARGS_${PKG_NAME}"
       echo_run ./configure --with-gaproot="$GAPROOT" $CONFIGFLAGS ${!CONFIG_ARGS_FLAG_NAME}
+      # hack: run `make clean` in case the package was built before with different settings
       echo_run "$MAKE" clean
     else
       echo_run ./configure "$GAPROOT"
+      # hack: run `make clean` in case the package was built before with different settings
```

Comment:
I don't understand this code. At least for Browse and EDIM it would be enough to just call `./configure "$GAPROOT"` (a `"$MAKE" clean` is not needed). This generates a new `Makefile` and this causes that `"$MAKE"` below will recompile the object file.

Reply:
yes but there are other package whose build system is not as clever; if one does `./configure` for a second time but with different arguments (or with a GAP that has changed to a different configuration), they don't rebuild their kernel extension at all. I've been working on getting those packages to fix / improve their build system, but until all are complete, we inserted as a workaround a `make clean` call to ensure they really rebuild. Which works great with all packages except edim and browse; hence we call configure again as a workaround for the new problem introduced by the first workaround.